contracts/programs/bubblegum/src/lib.rs

```diff
@@ -351,62 +550,85 @@ pub mod bubblegum {
+        Ok(())
+    }
+
+    pub fn approve_mint_authority_request(
+        ctx: Context<ApproveMintRequest>,
+        num_mints_to_approve: u64,
+    ) -> Result<()> {
+        let authority = &mut ctx.accounts.tree_authority;
+        let request = &mut ctx.accounts.mint_authority_request;
+        // Check that there are enough valid mints left in tree to approve
+        assert_enough_mints_to_approve(&authority, num_mints_to_approve)?;
+        authority.approve_mint_capacity(num_mints_to_approve);
+        request.approve(num_mints_to_approve)?;
+        Ok(())
+    }
+
+    pub fn close_mint_request(ctx: Context<CloseMintRequest>) -> Result<()> {
+        let authority = &mut ctx.accounts.tree_authority;
+        let request = &ctx.accounts.mint_authority_request;
+        // Transfer remaining mint capacity to authority
+        authority.restore_mint_capacity(request.num_mints_approved);
+        Ok(())
+    }
+
+    pub fn set_tree_delegate(ctx: Context<SetTreeDelegate>) -> Result<()> {
+        ctx.accounts.tree_authority.delegate = ctx.accounts.new_delegate.key();
+        Ok(())
+    }
+
     pub fn mint_v1(ctx: Context<MintV1>, message: MetadataArgs) -> Result<()> {
         // TODO -> Pass collection in check collection authority or collection delegate authority signer
         // TODO -> Separate V1 / V1 into seperate instructions
-        assert_metadata_is_mpl_compatible(&message)?;
         let owner = ctx.accounts.owner.key();
         let delegate = ctx.accounts.delegate.key();
-        let merkle_slab = ctx.accounts.merkle_slab.to_account_info();
+        let mint_authority = &mut ctx.accounts.mint_authority;
+        let merkle_slab = &ctx.accounts.merkle_slab;
 
-        // @dev: seller_fee_basis points is encoded twice so that it can be passed to marketplace instructions, without passing the entire, un-hashed MetadataArgs struct
-        let metadata_args_hash = keccak::hashv(&[message.try_to_vec()?.as_slice()]);
-        let data_hash = keccak::hashv(&[
-            &metadata_args_hash.to_bytes(),
-            &message.seller_fee_basis_points.to_le_bytes(),
-        ]);
-        let nonce = &mut ctx.accounts.authority;
-        let creator_data = message
-            .creators
-            .iter()
-            //TODO include verified
-            .map(|c| [c.address.as_ref(), &[c.share]].concat())
-            .collect::<Vec<_>>();
-        let creator_hash = keccak::hashv(
-            creator_data
-                .iter()
-                .map(|c| c.as_slice())
-                .collect::<Vec<&[u8]>>()
-                .as_ref(),
-        );
-        let asset_id = get_asset_id(&merkle_slab.key(), nonce.count);
-        let leaf = LeafSchema::new_v0(
-            asset_id,
+        // The mint authority must sign if it is not equal to the tree authority 
+        if mint_authority.key() != ctx.accounts.authority.key() {
+            assert!(mint_authority.is_signer);
```

Comment:
is this to enable the `tree_creator` to make some amount of NFTs freely available for open-season minting, no signing required by the user?

Reply:
Yes. I believe @austbot requested this as a possibility